main/src/ca/uwaterloo/flix/language/errors/NameError.scala

```diff
@@ -308,4 +308,16 @@ object NameError {
     def loc: SourceLocation = loc1 min loc2
   }
 
+  case class IllegalSignature(loc: SourceLocation) extends NameError {
```

Comment:
Add documentation and ensure that the error messages follows the "structure" of other error messages :)